### PR TITLE
[feat] add profile context and provider

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Added `src/utils/ProfileProvider.tsx` to create the context and context provider.
- Check comments at the bottom of the file for usage

Modified `src/api/supabase/queries/profiles.ts` to include a helper function to fetch profile by id.

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Review in any order.

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar

### CODE REVIEW

File: src/utils/ProfileProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+      const data = await fetchProfileById(sessionData.session.user.id as UUID);
+      setProfileData(data);
+    })();
+  }, []);
+
+  const providerValue = useMemo(
+    () => ({
+      profileData,
+      updateProfile: setProfileData,
```
Comment (pragyakallanagoudar) [inline src/utils/ProfileProvider.tsx]: In the pseudocode on the Linear issue, the `updateProfile` function looks like this:

```
// Update the profile data locally and trigger an API call to update it on Supabase
  const updateProfile = async (newProfileData: Profile) => {
    // Update the local state
    setProfileData(newProfileData);

    // [pragya] call the supabase query you wrote to update the profile for this user
  };
```

Wondering if we might be missing the last part here— i.e., when `profileData` is updated, is the corresponding entry in the `profiles` table on Supabase also updated?

On the settings page, when an edit is made to the profile, I think we ideally want to be able to call the `updateProfile` in the `ProfileContext` and have both the `Profile` object in the context and the entry on Supabase be updated.

File: src/utils/ProfileProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  );
+
+  return (
+    <ProfileContext.Provider value={providerValue}>
+      {children}
+    </ProfileContext.Provider>
+  );
+}
+
+/**
```
Comment (pragyakallanagoudar) [inline src/utils/ProfileProvider.tsx:67]: you're a g for this, this is so helpful

File: src/utils/ProfileProvider.tsx (hunk 1 above)
Comment (jinkang-0) [reply src/utils/ProfileProvider.tsx]: oh yes thanks for the catch!

**Review by pragyakallanagoudar — APPROVED** [review]
Looks great, thanks for responding to my feedback Jinkang! And thanks for contributing to profiles work, it is so valuable to us :)) 

This will probably change...a lot...once schema changes land, but we shall ❌ that 🌉 later❗

**Comment by varortz** [comment]
Looks good!! It was so useful reading up on useContext and Context providers to review this PR :)

**Review by varortz — APPROVED** [review]
Looks good!! It was so useful reading up on useContext and Context providers to review this PR :)

**Pull request merged**